### PR TITLE
Intel Vtune tracing integration

### PR DESCRIPTION
**Pull request by waahm7**
*Description of changes:*
- Adds Intel Vtune header and Option

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/aws/common/config.h.in [hunk 1]
```diff
@@ -19,5 +19,6 @@
 #cmakedefine AWS_HAVE_EXECINFO
 #cmakedefine AWS_HAVE_WINAPI_DESKTOP
 #cmakedefine AWS_HAVE_LINUX_IF_LINK_H
+#cmakedefine INTEL_NO_ITTNOTIFY_API
```
Comment (graebm) [inline include/aws/common/config.h.in:22]: we talked, it's probably simpler to have AwsCFlags.c add INTEL_NO_ITTNOTIFY_API to AWS_C_DEFINES_PRIVATE

File: THIRD-PARTY-LICENSES.txt [hunk 2]
```diff
@@ -0,0 +1,33 @@
+** ittapi ittnotify.h; version v3.24.2 -- https://github.com/intel/ittapi/blob/master/include/ittnotify.h
+/*
+  Copyright (C) 2005-2019 Intel Corporation
+
+  SPDX-License-Identifier: GPL-2.0-only OR BSD-3-Clause
+*/
+
+Copyright (c) <year> <owner>. All rights reserved.
```
[the remark this answers is not in the document]
Comment (waahm7) [reply THIRD-PARTY-LICENSES.txt]: Thanks, I have filled it.

File: include/aws/common/_ittnotify_private.h [hunk 3]
```diff
@@ -0,0 +1,4671 @@
...
+*/
+#ifndef _ITTNOTIFY_H_
+#define _ITTNOTIFY_H_
+/*
+ * This file has been not modified from its original version apart from adding the NOLINTBEGIN and NOLINTEND.
+ * This file is not intended for public usage.
+ */
+/* clang-format off */
+/* NOLINTBEGIN */
+
```
[the remark this answers is not in the document]
Comment (waahm7) [reply include/aws/common/_ittnotify_private.h]: Thanks, I have removed all the modifications from the file.

File: include/aws/common/config.h.in (hunk 1 above)
Comment (waahm7) [reply include/aws/common/config.h.in]: Thanks, fixed.

File: include/aws/common/_ittnotify_private.h [hunk 4]
```diff
@@ -0,0 +1,4671 @@
+/*
```
[the remark this answers is not in the document]
Comment (waahm7) [reply include/aws/common/_ittnotify_private.h]: Thanks, moved it to external folder.

File: CMakeLists.txt [hunk 5]
```diff
@@ -41,6 +41,9 @@ file(GLOB AWS_COMMON_HEADERS
 file (GLOB AWS_COMMON_EXTERNAL_HEADERS
         "include/aws/common/external/*.h")
 
+file (GLOB AWS_COMMON_EXTERNAL_INSTALLED_HEADERS
```
Comment (waahm7) [inline CMakeLists.txt:44]: If you have any suggestions for a better name for this variable, please feel free to share them.

Comment (graebm) [reply CMakeLists.txt:44]: I like it

File: cmake/AwsCFlags.cmake [hunk 6]
```diff
@@ -250,6 +251,11 @@ function(aws_set_common_properties target)
         endif()
     endif()
 
+    if(AWS_ENABLE_TRACING)
+        target_link_libraries(${target} PRIVATE ittnotify)
+    else()
+        list(APPEND AWS_C_DEFINES_PRIVATE -DINTEL_NO_ITTNOTIFY_API) # Disable intel notify api if tracing is not enabled
```
Comment (graebm) [inline cmake/AwsCFlags.cmake:257]: utterly trivial: kind of a long comment to tack onto the end of a line
```suggestion
        # Disable intel notify api if tracing is not enabled
        list(APPEND AWS_C_DEFINES_PRIVATE -DINTEL_NO_ITTNOTIFY_API)
```

**Review by graebm — APPROVED** [review]

**Pull request merged**